ironfish-cli/src/utils/chainport/utils.ts

```diff
@@ -0,0 +1,111 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { RpcWalletTransaction, TransactionType } from '@ironfish/sdk'
+import { getConfig } from './config'
+import { ChainportMemoMetadata } from './metadata'
+
+type ChainportTransactionDetails = {
+  chainportNetworkId: number
+  address: string
+}
+
+type NotChainportTransaction = {
+  isChainportTransaction: false
+}
```

Comment:
nit -- this looks a little funny to wrap `false` this way

Do you think it would be simpler to have a function that returns null or undefined if it's not a Chainport transaction and returns an object with details otherwise?

Reply:
Sounds good, I changed it to be undefined if not a chainport transaction